src/Transaction.js

```diff
@@ -3,12 +3,38 @@
  * An account of what has happened.
  */
 
-export default function Transaction (type, payload, complete) {
-  return {
-    type     : `${ type }`,
-    error    : false,
-    active   : arguments.length > 1,
-    payload  : payload,
-    complete : complete
+import coroutine  from './coroutine'
+import flatten    from './flatten'
+import eventually from './eventually'
+
+export default function Transaction (action, payload) {
+  this.action  = action
+  this.type    = `${ action }`
+  this.active  = arguments.length > 1
+  this.payload = payload
+}
+
+Transaction.prototype = {
+  action   : null,
+  active   : false,
+  error    : false,
+  payload  : undefined,
```

Comment:
Yes. I want to keep the shape and type information for transactions
consistent for better V8 optimization. These these things will be operated
on pretty frequently for certain use cases.

On Tue, Jan 26, 2016 at 11:23 AM, Lawson Kurtz <notifications@github.com>
wrote:

> In src/Transaction.js
> <https://github.com/vigetlabs/microcosm/pull/94#discussion_r50858060>:
>
> > +import coroutine  from './coroutine'
> > +import flatten    from './flatten'
> > +import eventually from './eventually'
> > +
> > +export default function Transaction (action, payload) {
> > +  this.action  = action
> > +  this.type    = `${ action }`
> > +  this.active  = arguments.length > 1
> > +  this.payload = payload
> > +}
> > +
> > +Transaction.prototype = {
> > +  action   : null,
> > +  active   : false,
> > +  error    : false,
> > +  payload  : undefined,
>
> need this?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/vigetlabs/microcosm/pull/94/files#r50858060>.
>